### PR TITLE
[RTM] Add the TemplateInheritance::getTemplatePath() method (see #218).

### PR DESCRIPTION
**Pull request by ausi**
As discussed in #218

### CODE REVIEW

File: src/Resources/contao/library/Contao/TemplateInheritance.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function inherit()
 		while ($this->strParent !== null)
 		{
 			$strCurrent = $this->strParent;
-			$strParent = $this->strDefault ?: \Controller::getTemplate($this->strParent, $this->strFormat);
+			$strParent = $this->strDefault ?: static::getTemplatePath($this->strParent, $this->strFormat);
```
Comment (leofeyer) [inline src/Resources/contao/library/Contao/TemplateInheritance.php]: Why did you make the method static? Does `$this->getTemplatePath()` not work?

Comment (ausi) [inline src/Resources/contao/library/Contao/TemplateInheritance.php]: `$this->getTemplatePath()` would work. Should I change only the call to `$this->getTemplatePath()` or should I also make the method non-static?

Comment (leofeyer) [inline src/Resources/contao/library/Contao/TemplateInheritance.php]: Please also make the method non-static.

Comment (ausi) [inline src/Resources/contao/library/Contao/TemplateInheritance.php]: Done in 7cdf8d7069487b6d7ed637321eec9ab327cc8e27

**Comment by aschempp** [comment]
The method should be called `getTemplate` to be BC. That works totally fine, see http://php.net/manual/en/language.oop5.traits.php#language.oop5.traits.precedence

**Comment by leofeyer** [comment]
Then the method would override `Controller::getTemplate()`, wouldn't it?

**Comment by aschempp** [comment]
Yes it would. But that's totally fine, as it calls `\Controller::getTemplate` internally.

**Comment by leofeyer** [comment]
Not sure. @ausi What do you think?

**Comment by ausi** [comment]
As I commented in #218

> > But we could simply give the method another name :)

> It wouldn’t be backwards compatible then but much cleaner than overwriting a method of a parent class with a trait I guess :)

I think overwriting a method of the parent class and calling that parent class from there could be confusing. IMO the question is if we want it clean or backwards compatible.

**Comment by leofeyer** [comment]
+1 for clean.

**Comment by aschempp** [comment]
It's legacy code, it does not need to be "clean", it needs to be BC ;-)
According to the PHP docs, it would be totally valid to do…

**Pull request merged**